runtime/metricsview/astsql.go

```diff
@@ -244,6 +262,15 @@ func (b *sqlBuilder) writeJoin(joinType JoinType, baseSelect, joinSelect *Select
 		}
 		lhs := b.ast.sqlForMember(baseSelect.Alias, f.Name)
 		rhs := b.ast.sqlForMember(joinSelect.Alias, f.Name)
+		if comp && f.Time {
+			intv, err := b.ast.interval()
+			if err != nil {
+				return err
+			}
+
+			// example: base.ts IS NOT DISTINCT FROM comparison.ts - (? - ?) -- <comparison-start> - <base-start>
+			rhs = fmt.Sprintf("(%s - INTERVAL %s MILLISECONDS)", rhs, intv)
+		}
```

Comment:
I think we can move this into the `ast.addTimeComparisonMeasure` function, where we could add the subtraction to the dimension expression in the comparison select instead, around here: https://github.com/rilldata/rill/blob/ec8712134a6c03c0aa5a1ef4b6b43ebbf4c24a77/runtime/metricsview/ast.go#L803-L807

This would allow us to keep the regular `IS NOT DISTINCT FROM` check here, and avoid accessing the `ast` too much from the SQL builder phase.